action/protocol/staking/staking_statereader.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2023 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/iotexproject/iotex-proto/golang/iotexapi"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+type (
+
+	// compositiveStakingStateReader is the compositive staking state reader, which combine native and liquid staking
+	compositiveStakingStateReader struct {
+		liquidIndexer LiquidStakingIndexer
+		nativeIndexer *CandidatesBucketsIndexer
+		nativeSR      CandidateStateReader
+	}
+)
+
+var (
+	_ ReadState = (*compositiveStakingStateReader)(nil)
+)
+
+// newCompositiveStakingStateReader creates a new compositive staking state reader
+func newCompositiveStakingStateReader(liquidIndexer LiquidStakingIndexer, nativeIndexer *CandidatesBucketsIndexer, nativeSR CandidateStateReader) *compositiveStakingStateReader {
```

Comment:
does this class share same interfaces with another class?

Reply:
> does this class share same interfaces with another class? if yes, a private interface would be better as the returned

it should be a `ReadState`